File: src/test/java/com/ealanta/productapp/item/SwatchDeSerializationTest.java
package com.ealanta.productapp.item;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.core.io.Resource;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.util.StreamUtils;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

@RunWith(SpringRunner.class)
public class SwatchDeSerializationTest {

    private static final Logger LOGGER = LoggerFactory.getLogger(SwatchDeSerializationTest.class);

    @Value("classpath:input/swatch.json")
    private Resource swatch;

    private String swatchJson;

    @Before
    public void setup() throws IOException {
        this.swatchJson = StreamUtils.copyToString(swatch.getInputStream(), StandardCharsets.UTF_8);
        LOGGER.debug(this.swatchJson);
    }

    @Test
    public void testJsonToSwatch() throws Exception {
        ObjectMapper mapper = new ObjectMapper();
        ColorSwatchItem swatch = mapper.readValue(this.swatchJson, ColorSwatchItem.class);
        Assert.assertNotNull(swatch);
        Assert.assertEquals("237494589", swatch.getSkuId());
        Assert.assertEquals("Red", swatch.getBasicColor());
        Assert.assertEquals("Wine", swatch.getColor());
        LOGGER.debug(swatch.toString());
    }

}

File: src/src/test/kotlin/com/ealanta/productapp/item/SwatchDeSerializationTest.kt
package com.ealanta.productapp.item;

import com.ealanta.productapp.domain.ColorSwatchItem
import com.ealanta.productapp.domain.PriceDetails
import com.fasterxml.jackson.databind.node.TextNode
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import mu.KotlinLogging
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Value
import org.springframework.core.io.Resource
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.util.StreamUtils
import java.nio.charset.StandardCharsets
import kotlin.properties.Delegates
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

@RunWith(SpringRunner::class)
class SwatchDeSerializationTest {

    private val logger = KotlinLogging.logger {}

    @Value("classpath:input/swatch.json")
    lateinit var swatch: Resource

    var swatchJson by Delegates.notNull<String>()

    @Before
    fun setup() {
        this.swatchJson = StreamUtils.copyToString(swatch.getInputStream(), StandardCharsets.UTF_8);
        logger.debug(this.swatchJson);
    }

    @Test
    fun testJsonToSwatch() {
        val mapper = jacksonObjectMapper()
        val swatch = mapper.readValue<ColorSwatchItem>(this.swatchJson)
        assertNotNull(swatch);
        assertEquals("237494589", swatch.skuId);
        assertEquals("Red", swatch.basicColor);
        assertEquals("Wine", swatch.color);
        logger.debug(swatch.toString());
    }

    @Test
    fun testSerialization() {
        val mapper = jacksonObjectMapper()
        val details = PriceDetails().apply{
            was="WAS"
            currency="GBP"
            then1="THEN1"
            then2="THEN2"
            now = TextNode("NOW")
        }
        print(details)
        val json = mapper.writeValueAsString(details);
        print(json)

        val pricedDetails2 = mapper.readValue<PriceDetails>(json)
        println(pricedDetails2)
    }
}
